lib/entry.js

```diff
@@ -94,100 +80,42 @@ Ep.runModuleGetters = function (module) {
+  var getter = this.getters[name];
+
+  if (getter) {
     try {
       // Update module.exports[name] with the current value so that CommonJS
       // require calls remain consistent with module.importSync.
-      return module.exports[name] =
-        this.getters[name].call(module);
-
-    } catch (e) {
-      // If the getter threw an exception, avoid updating module.exports
-      // and return undefined.
-    }
+      module.exports[name] = getter();
+    } catch (e) {}
   }
 };
 
 // Called whenever module.exports might have changed, to trigger any
 // setters associated with the newly exported values.
 Ep.runModuleSetters = function (module) {
-  var entry = this;
-  var names = Object.keys(this.setters);
-
   // Make sure module.exports is up to date before we call
   // module.getExportByName(name).
   this.runModuleGetters(module);
 
-  if (! names.length) {
-    ++this.runCount;
-    return;
-  }
-
-  // Invoke the given callback once for every (setter, value, name) triple
-  // that needs to be called. Note that forEachSetter does not call any
-  // setters itself, only the given callback.
-  function forEachSetter(callback, context) {
-    var nameCount = names.length;
-
-    for (var i = 0; i < nameCount; ++i) {
-      var name = names[i];
-      var setters = entry.setters[name];
-      var keys = Object.keys(setters);
-      var keyCount = keys.length;
-
-      for (var j = 0; j < keyCount; ++j) {
-        var key = keys[j];
-        var value = module.getExportByName(name);
-
-        if (name === "*") {
-          var valueNames = Object.keys(value);
-          var valueNameCount = valueNames.length;
-
-          for (var k = 0; k < valueNameCount; ++k) {
-            var valueName = valueNames[k];
-            call(setters[key], value[valueName], valueName);
-          }
-
-        } else {
-          call(setters[key], value, name);
-        }
-      }
-    }
-
-    function call(setter, value, name) {
-      if (name === "__esModule") {
-        // Ignore setters asking for module.exports.__esModule.
-        return;
-      }
-
-      setter.last = setter.last || Object.create(null);
-
-      if (! hasOwn.call(setter.last, name) ||
-          setter.last[name] !== value) {
-        // Only invoke the callback if we have not called this setter
-        // (with a value of this name) before, or the current value is
-        // different from the last value we passed to this setter.
-        return callback.apply(context, arguments);
-      }
-    }
-  }
-
   // Lazily-initialized object mapping parent module identifiers to parent
   // module objects whose setters we might need to run.
   var relevantParents;
 
   // Take snapshots of setter.parent.exports for any setters that we are
   // planning to call, so that we can later determine if calling the
   // setters modified any of those exports objects.
-  forEachSetter(function (setter, value, name) {
+  forEachSetter(module, this, function (setter, name, value) {
```

Comment:
common setter order is `name` of the thing you want to set then the `value` of the thing to set. The `setter` bit is a backwards from convention so it gets a comment just below.